Sources/UID2/UID2Manager.swift

```diff
@@ -97,27 +97,48 @@ public final actor UID2Manager {
     
     // iOS Way to Provide Initial Setup from Outside
     // Web Way --> https://github.com/IABTechLab/uid2-web-integrations/blob/5a8295c47697cdb1fe36997bc2eb2e39ae143f8b/src/uid2Sdk.ts#L153-L154
+    
+    /// Set UID2 Identity for UID2Manager to manage
+    /// - Parameter identity: UID2 Identity for UID2Manager to manage
     public func setIdentity(_ identity: UID2Identity) async {
         if let validatedIdentity = await validateAndSetIdentity(identity: identity, status: nil, statusText: nil) {
             await triggerRefreshOrSetTimer(validIdentity: validatedIdentity)
         }
     }
-
+    
+    /// Reset UID2 Identity state in UID2Manager
     public func resetIdentity() async {
         self.identity = nil
         self.identityStatus = .noIdentity
         KeychainManager.shared.deleteIdentityFromKeychain()
     }
     
+    /// Manually Refresh UID2 Identity
     public func refreshIdentity() async {
         guard let identity = identity else {
             return
         }
         await refreshToken(identity: identity)
     }
     
-    public func getAdvertisingToken() -> String? {
-        return self.identity?.advertisingToken
+    /// Get Advertising Token if Valid
+    /// - Returns: Adversting Token if valid, nil if not valid
+    public func getAdvertisingToken() async -> String? {
+        
+        guard let identity = self.identity else {
+            self.identityStatus = .noIdentity
+            return nil
+        }
+
+        let identityPackage = await getIdentityPackage(identity: identity)
```

Comment:
I'm trying to better understand in what scenario will the Identity and IdentityStatus be out of sync, so that this additional call to `getIdentityPackage` is required. From my understanding:
 - When we load the identity from disk, we should update the status to reflect it's de-serialised form
 - When we refresh the identity (after `refreshFrom`), we update the status to reflect it's new status
 - The iOS code is checking periodically (by default every 5 seconds) to catch a time after `refreshFrom` which should be much earlier than it expires, so we refresh.

In what scenario can these be out of sync? If it's possible, it might be best to start with a unit test to indicate the issue we're trying to solve.

Reply:
I've now refactored this to mirror Android's Job (Task in Swift) approach to refreshing data and updating state in `UID2Manager`.  Resolving comment as it's now out of date.